src/Visualization/Python/GenerateXdmf.py

```diff
@@ -29,6 +29,10 @@ def generate_xdmf(file_prefix, output, subfile_name, start_time, stop_time,
         file_prefix)
 
     element_data = h5files[0][0].get(subfile_name + '.vol')
+    if element_data is None:
+        raise ValueError(("Could not open subfile name '{}.vol'. Available "
+                          "subfiles: {}").format(subfile_name,
+                                                 h5files[0][0].keys()))
```

Comment:
Looks good. I suggest you also add this test to `Test_GenerateXdmf.py`, e.g. before line 19:

```py
with self.assertRaisesRegex(ValueError, 'Could not open subfile'):
    generate_xdmf( # ... copy from line 19, change subfile_name to something wrong
```

(this will also give an example how to test exceptions in Py code for future reference)

Reply:
apparently the code doesn't :shrug:
```
line 40, in test_subfile_not_found
    with self.assertRaisesRegex(ValueError, 'Could not open subfile'):
AttributeError: 'TestGenerateXdmf' object has no attribute 'assertRaisesRegex'
```
could this be a python2 vs 3 issue?

Reply:
Oh right it was renamed `assertRaisesRegexp` -> `assertRaisesRegex` in Py3.2 (note trailing `p`) 🙄.. add this below the imports:

```py
# For Py2 compatibility 
try:
     unittest.TestCase.assertRaisesRegex
except AttributeError:
    unittest.TestCase.assertRaisesRegex = unittest.TestCase.assertRaisesRegexp

```

Reply:
Yay that worked! Thank you so much :D